### PR TITLE
[REVIEW] Fix public headers including non-public headers

### PR DESCRIPTION
**Pull request by trevorsm7**
Closes https://github.com/rapidsai/cudf/issues/3312

### CODE REVIEW

**Comment by harrism** [comment]
Looks like you missed one? :)

**Comment by trevorsm7** [comment]
Oops, gather.cuh got merged ahead and broke a few things. Looks like it has new dependencies on legacy headers for some reason.

File: cpp/include/cudf/detail/utilities/cuda.cuh [hunk 1]
```diff
@@ -103,6 +104,43 @@ __device__ T single_lane_block_sum_reduce(T lane_value) {
...
 }
 
+/**
+ * @brief for each warp in the block do a reduction (summation) of the
+ * `__popc(bit_mask)` on a certain lane (default is lane 0).
+ * @param[in] bit_mask The bit_mask to be reduced.
+ * @return[out] result of each block is returned in thread 0.
+ */
+template <class bit_container, int lane = 0>
+__device__ __inline__ size_type single_lane_popc_block_reduce(bit_container bit_mask) {
```
Comment (harrism) [inline cpp/include/cudf/detail/utilities/cuda.cuh]: Can you rename this function `single_lane_block_popc_reduce` to match `single_lane_block_sum_reduce`?

Comment (trevorsm7) [reply cpp/include/cudf/detail/utilities/cuda.cuh]: Done

**Pull request merged**